pkg/snapshot/restorer/restorer.go

```diff
@@ -430,23 +455,29 @@ func (r *Restorer) applyDeltaSnapshots(clientKV client.KVCloser, ro brtypes.Rest
 	}
 
 	var (
-		remainingSnaps  = snapList[1:]
-		numSnaps        = len(remainingSnaps)
-		numFetchers     = int(math.Min(float64(numMaxFetchers), float64(numSnaps)))
-		snapLocationsCh = make(chan string, numSnaps)
-		errCh           = make(chan error, numFetchers+1)
-		fetcherInfoCh   = make(chan brtypes.FetcherInfo, numSnaps)
-		applierInfoCh   = make(chan brtypes.ApplierInfo, numSnaps)
-		stopCh          = make(chan bool)
-		wg              sync.WaitGroup
+		remainingSnaps    = snapList[1:]
+		numSnaps          = len(remainingSnaps)
+		numFetchers       = int(math.Min(float64(numMaxFetchers), float64(numSnaps)))
+		snapLocationsCh   = make(chan string, numSnaps)
+		errCh             = make(chan error, numFetchers+1)
+		fetcherInfoCh     = make(chan brtypes.FetcherInfo, numSnaps)
+		applierInfoCh     = make(chan brtypes.ApplierInfo, numSnaps)
+		wg                sync.WaitGroup
+		stopCh            = make(chan bool)
+		stopHandleAlarmCh = make(chan bool)
+		dbSizeAlarmCh     = make(chan string)
+		dbSizeDisAlarmCh  = make(chan bool)
```

Comment:
I feel we are increasing complexity by adding these many channels. I understand that the purpose of `HandleAlarm()` and these channels is to help catch an alarm from etcd and then disarm it. But do we need to do this so proactively is my question. Let's say we are applying a delta snapshot and suddenly the threshold is breached, and an etcd alarm is raised. By the time the alarm is handled and disarmed asynchronously via these channels, many delta events would have failed to apply, right?

Instead, we anyway already have `periodicallyMakeEtcdLean`, which ensures that the etcd is compacted+defragmented after applying every X delta snapshots. I think this should suffice for now, rather than trying to handle alarms that are raised when the DB size breaches the quota.

The general solution for this problem is to use DB-size based defragmentation, which will be coming in steward, and can be switched on for restoration as well as for regular etcd functioning. Once we have this, then we don't need any special handling for restoration case. WDYT?

Reply:
> these channels is to help catch an alarm from etcd and then disarm it.

nopes, you have misunderstood the code. Please refer to this comment: https://github.com/gardener/etcd-backup-restore/pull/668#issuecomment-1818283375
Let me know if you have doubts.